### ISSUE TITLE
Usability improvment for ``use-a-generator`` in test assertion

### ISSUE THREAD

**Issue by Pierre-Sassoulas** [pylint-dev/pylint#9125]
### Current problem

Right now the following code raises a ``use-a-generator`` warning:

```python
def test_use_a_generator():
    assert all([int(i) == i for i in [1, 2, 3, "4"]])
```

Result of ``pytest a.py``:

```python
_____________ test_use_a_generator _____________ 
    def test_use_a_generator():
>       assert all([int(i) == i for i in [1, 2, 3, "4"]])
E       assert False
E        +  where False = all([True, True, True, False])
```

The fix is:
```python
def test_use_a_generator():
    assert all(int(i) == i for i in [1, 2, 3, "4"])
```
But the result of ``pytest a.py`` becomes less clear:

```python
_____________ test_use_a_generator _____________ 
    def test_use_a_generator():
>       assert all(int(i) == i for i in [1, 2, 3, "4"])
E       assert False
E        +  where False = all(<generator object test_use_a_generator.<locals>.<genexpr> at 0x7fd1b67eaf80>)
```
And the expected performance improvement is not easy to defend because most of the time the test should pass so all the values will have to be calculated. When the test fail the first thing to do will be to revert the changes

### Desired solution

When a ``use-a-generator`` should be raised in an assert node (inside a code construct detected as a test by pytest/unittest -- named ``test_`` or test classes?) do not raise  ? ``consider-using-a-generator`` raised instead ? Do nothing to keep it simple and stupid ?
